plugins/ipv6only/plugin.go

```diff
@@ -0,0 +1,78 @@
+// Copyright 2018-present the CoreDHCP Authors. All rights reserved
+// This source code is licensed under the MIT license found in the
+// LICENSE file in the root directory of this source tree.
+
+package ipv6only
+
+// This plugin implements RFC8925: if the client has requested the
+// IPv6-Only Preferred option, then add the option response and then
+// terminate processing immediately.
+//
+// This module should be invoked *before* any IP address
+// allocation has been done, so that the yiaddr is 0.0.0.0 and
+// no pool addresses are consumed for compatible clients.
+//
+// The optional argument is the V6ONLY_WAIT configuration variable,
+// as described in RFC8925 section 3.2.
+
+import (
+	"bytes"
+	"encoding/binary"
+	"errors"
+	"strconv"
+
+	"github.com/coredhcp/coredhcp/handler"
+	"github.com/coredhcp/coredhcp/logger"
+	"github.com/coredhcp/coredhcp/plugins"
+	"github.com/insomniacslk/dhcp/dhcpv4"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	// Not yet defined in dhcpv4/types.go
+	OptionV6OnlyPreferred dhcpv4.GenericOptionCode = 108
+)
+
+var log = logger.GetLogger("plugins/ipv6only")
+
+var v6only_wait uint32
+
+var Plugin = plugins.Plugin{
+	Name:   "ipv6only",
+	Setup4: setup4,
+}
+
+func setup4(args ...string) (handler.Handler4, error) {
+	if len(args) > 0 {
+		v, err := strconv.Atoi(args[0])
+		if err != nil {
+			return nil, err
+		}
+		if v < 0 || v > 0x7fffffff {
```

Comment:
The reason is because strconv.Atoi returns a signed int.

In Go, int can be [either 32 or 64 bits](https://go.dev/ref/spec#Numeric_types). On a platform where int is 32 bits, then I'm pretty sure that the following would [fail to compile](https://go.dev/play/p/RYWircld6MH):

```
		if v < 0 || v > 0xffffffff {
```

If there were Atoi64 or ParseInt64 then I would use it, but I don't think there is.

Given that 2<sup>31</sup>-1 seconds is a very long time, I dodged the issue by making the upper limit as shown.

Reply:
This problem goes away if https://github.com/insomniacslk/dhcp/pull/524 is accepted, and this option becomes a time.Duration

Reply:
If that upstream `time.Duration` change doesn't go through (although there's no reason to think it wouldn't), [`strconv.ParseInt`](https://pkg.go.dev/strconv#ParseInt) returns an `int64`, and there's also `strconv.ParseUint`. But let's go with Duration if we can

Reply:
That will teach me to read the function signatures and not just the function names :-)